packages/story-editor/src/app/media/utils/useMediaUploadQueue/index.js

```diff
@@ -207,52 +203,226 @@ function useMediaUploadQueue() {
+        currentTranscodingItem.current = null;
+      }
+    },
+    [startTranscoding, finishTranscoding, convertGifToVideo, cancelUploading]
+  );
+
+  const trimVideoItem = useCallback(
+    async (item) => {
+      const { id, file, additionalData = {}, trimData } = item;
+
+      startTrimming({ id });
+
+      try {
+        const newFile = await trimVideo(file, trimData.start, trimData.end);
+
+        if (!isMounted.current) {
+          return;
+        }
+
+        additionalData.meta = {
+          ...additionalData.meta,
+          web_stories_trim_data: trimData,
+        };
+        finishTrimming({ id, file: newFile, additionalData });
+      } catch (error) {
+        // Cancel uploading if there were any errors.
+        cancelUploading({ id, error });
+
+        trackError('upload_media', error?.message);
+      } finally {
+        currentTranscodingItem.current = null;
+      }
+    },
+    [startTrimming, finishTrimming, trimVideo, cancelUploading]
+  );
+
+  const muteVideoItem = useCallback(
+    async (item) => {
+      const { id, file, additionalData = {} } = item;
+
+      startMuting({ id });
+
+      try {
+        const newFile = await stripAudioFromVideo(file);
+
+        if (!isMounted.current) {
+          return;
+        }
+
+        additionalData.web_stories_is_muted = true;
+        finishMuting({ id, file: newFile, additionalData });
+      } catch (error) {
+        // Cancel uploading if there were any errors.
+        cancelUploading({ id, error });
+
+        trackError('upload_media', error?.message);
+      } finally {
+        currentTranscodingItem.current = null;
+      }
+    },
+    [startMuting, finishMuting, stripAudioFromVideo, cancelUploading]
+  );
+
+  const optimizeVideoItem = useCallback(
+    async (item) => {
+      const { id, file, additionalData = {} } = item;
+
+      startTranscoding({ id });
+
+      try {
+        const newFile = await transcodeVideo(file);
+
+        if (!isMounted.current) {
+          return;
+        }
+
+        additionalData.web_stories_media_source = 'video-optimization';
+        finishTranscoding({ id, file: newFile, additionalData });
+      } catch (error) {
+        // Cancel uploading if there were any errors.
+        cancelUploading({ id, error });
+
+        trackError('upload_media', error?.message);
+      } finally {
+        currentTranscodingItem.current = null;
+      }
+    },
+    [startTranscoding, finishTranscoding, transcodeVideo, cancelUploading]
+  );
+
+  const uploadVideo = useCallback(
+    async (item) => {
+      const { id, file, resource, additionalData = {} } = item;
+      let { posterFile } = item;
+
+      let newResource;
+
+      // The newly uploaded file won't have a poster yet.
+      // However, we'll likely still have one on file.
+      // Add it back so we're never without one.
+      // The final poster will be uploaded later by uploadVideoPoster().
+      newResource = await uploadFile(file, additionalData);
```

Comment:
Can't we just do this?
```suggestion
      let newResource = await uploadFile(file, additionalData);
```

Reply:
We can. Just an oversight from refactoring.